components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultRequestCoordinator.java

```diff
@@ -1004,7 +1004,10 @@ private void buildOutboundQueryString(HttpServletRequest request, Authentication
                     .append("&relyingParty=").append(URLEncoder.encode(context.getRelyingParty(), "UTF-8"))
                     .append("&type=").append(context.getRequestType()).append("&")
                     .append(FrameworkConstants.REQUEST_PARAM_SP).append("=")
-                    .append(URLEncoder.encode(context.getServiceProviderName(), "UTF-8")).append("&isSaaSApp=")
+                    .append(URLEncoder.encode(context.getServiceProviderName(), "UTF-8")).append("&")
+                    .append(FrameworkConstants.REQUEST_PARAM_SP_UUID).append("=")
+                    .append(context.getServiceProviderResourceId()).append("&")
```

Comment:
Since we use UUIDs as application id at the moment, and UUIDs have only URL-safe characters URL encoding is not needed now,
but for safe side (incase the resource id format will change in the feature) shall we URL encode this value?

Reply:
Addressed with https://github.com/wso2/carbon-identity-framework/pull/5361/commits/3f89468bb1ca538417209ca475430a458b562a9f